_rules/SC2-4-4-link-has-name.md

```diff
@@ -0,0 +1,157 @@
+---
+name: Links have an accessible name
+description: |
+  Each link has an accessible name
+
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects: # Remove what is not applicable
+- DOM Tree
+- CSS Styling
+
+authors:
+- Wilco Fiers
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any HTML or SVG element with the [semantic role](#semantic-role) of `link` that is [visible on the page](#visible-on-the-page) or [exposed to assistive technologies](#exposed-to-assistive-technologies).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is [non-empty](#non-empty).
+
+## Assumptions
+
+- The rule assumes that all links are user interface components as defined by WCAG 2. When the link role is used incorrectly, this assumption may not be true.
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=244#navigation-mechanisms-refs](https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=244#navigation-mechanisms-refs)
+- [ARIA7: Using aria-labelledby for link purpose](https://www.w3.org/TR/WCAG20-TECHS/ARIA7.html)
+- [ARIA8: Using aria-label for link purpose](https://www.w3.org/TR/WCAG20-TECHS/ARIA8.html)
+- [F89: Failure of Success Criteria 2.4.4, 2.4.9 and 4.1.2 due to using null alt on an image where the image is the only content in a link](http://www.w3.org/TR/WCAG20-TECHS/F89.html)
+
+## Test Cases
+
+### Passed
```

Comment:
Empty line after "passed" - also, this needs tests for:
- passing when link is off screen
- passing when link is not in accessibility tree